R/enrichDAVID.R

```diff
@@ -82,86 +80,68 @@ enrichDAVID <- function(gene,
     david.res <- addList(david, gene, idType=idType,
                          listName="clusterProfiler",
                          listType="Gene")
-
-
+    
+    
     if (david.res$inDavid == 0) {
         stop("All id can not be mapped. Please check 'idType' parameter...")
     }
-
+    
     if (!missing(universe)) {
         david.res <- addList(david, universe, idType=idType,
                              listName="universe",
                              listType="Background")
     }
-
+    
     setAnnotationCategories(david, annotation)
     x <- getFunctionalAnnotationChart(david, threshold=1, count=minGSSize)
-
-    if (length(x@.Data) == 0) {
-        warning("No significant enrichment found...")
-        return(NULL)
+    
+    if (!is.na(maxGSSize) && !is.null(maxGSSize)) {
+        x <- x[x$Count <= maxGSSize, ]
```

Comment:
filtered by background gene set size in other functions.

Reply:
Ummm, 我好像是理解错了，我一直以为`getFunctionalAnnotationChart(..., count=minGSsize)`里的count是在控制富集的用户基因集的最小数。。。是不是maxGSSize和minGSSize都应该过滤在富集的背景基因集大小上？所以应该是
```suggestion
if (!is.na(maxGSSize) && !is.null(maxGSSize)) {
  x <- x[x$Pop.Hits <= maxGSSize, ]
}
```